### PR TITLE
Add some information about a common error with Caffeine and native

### PR DESCRIPTION
**Pull request by gsmet**
Related to https://github.com/quarkusio/quarkus/discussions/22018

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]
LGTM. Added a personal comment which you may ignore if you like 😉

File: docs/src/main/asciidoc/cache.adoc [hunk 1]
```diff
@@ -748,3 +748,36 @@ When the call to the remote throws an exception, the cache does not store the re
...
+        at java.lang.Class.forName(DynamicHub.java:1408)
+        at com.github.benmanes.caffeine.cache.NodeFactory.newFactory(NodeFactory.java:111)
+        at com.github.benmanes.caffeine.cache.BoundedLocalCache.<init>(BoundedLocalCache.java:240)
+        at com.github.benmanes.caffeine.cache.SS.<init>(SS.java:31)
+        at com.github.benmanes.caffeine.cache.SSMS.<init>(SSMS.java:64)
+        at com.github.benmanes.caffeine.cache.SSMSA.<init>(SSMSA.java:43)
+----
+<1> `PSAMS` is one of the many cache implementation classes of Caffeine so this part may vary.
+
+When you encounter this error, you can easily fix it by adding the following annotation to any of your application classes:
```
Comment (gastaldi) [inline docs/src/main/asciidoc/cache.adoc:775]: This is my personal choice and I know that it's up to the developer to follow, but perhaps we could suggest adding the `RegisterForReflection` annotation to a single class in your application (eg. `Reflections`) for this purpose (eg. https://github.com/quarkusio/registry.quarkus.io/blob/main/src/main/java/io/quarkus/registry/app/Reflections.java)

Comment (gsmet) [reply docs/src/main/asciidoc/cache.adoc:775]: Well they can do whatever they want, I don't really care :).

Comment (gastaldi) [reply docs/src/main/asciidoc/cache.adoc:775]: I know, it's just a personal suggestion :)

Comment (gsmet) [reply docs/src/main/asciidoc/cache.adoc]: I added your alternative.

**Review by gastaldi — APPROVED** [review]

**Pull request merged**